### DISCUSSION TITLE
pyzbar

### DISCUSSION THREAD

**Discussion by YasseenAlhusammi** [flet-dev/flet#3563] [Packages] [answered]
Add pyzbar to flet please

**Comment by burhansvural** [comment]
> pyzbar

```python
# using opencv
import cv2
import flet as ft
from pyzbar.pyzbar import decode, ZBarSymbol


def main(page: ft.Page):
    txt_barcode_data = ft.Text()
    txt_barcode_type = ft.Text()
    barcode_image = ft.Image(
        width=350,
        height=200,
        src="barcodes.png"
    )
    page.add(
        barcode_image,
        txt_barcode_data,
        txt_barcode_type
    )
    image = cv2.imread("./assets/barcodes.png")
    detected_barcodes = decode(image)

    value: str = ""
    data: str = ""
    
    for barcode in detected_barcodes:
        (x, y, w, h) = barcode.rect
        cv2.rectangle(image, (x, y), (x + w, y + h), (255, 0, 0), 5)
        value += f"{barcode.data}\r\n"
        data += f"{barcode.type}\r\n"

    txt_barcode_data.value = value
    txt_barcode_type.value = data
    txt_barcode_data.update()
    txt_barcode_type.update()


ft.app(target=main, assets_dir="assets")

```

**Reply by YasseenAlhusammi** [reply]
This package is already on PyPI but contains external files that do not work on android systems with version flet==0.25.0.dev3519
https://github.com/flet-dev/flet/discussions/4158

**Comment by YasseenAlhusammi** [comment]
#4158

**Comment by FeodorFitsner** [comment]
Related discussion: https://github.com/flet-dev/flet/discussions/4163

**Reply by YasseenAlhusammi** [reply]
Please, please, start packing and give priority to this package. My project has been on hold for a year because of this package. I think it is easy for you, my dear brother.

**Comment by YasseenAlhusammi** [comment]
I don't know what I did. I built and the build succeeded as in the following link:
https://github.com/YasseenAlhusammi/build-lib-native/actions/runs/12208436486
, and I didn't understand what's next. Can I benefit from the build output or not? I don't know how to benefit from the build results.

**Comment by yasseenalhusammin** [comment]
I have successfully built the library as in the project: https://github.com/yasseenalhusammin/build-lib-native and the build results were as in the attachment https://github.com/yasseenalhusammin/build-lib-native/actions/runs/13710454805/artifacts/2707673124 as follows: Now I want to know how to include the build results (ready library) within the flet project, is there any help in this aspect? I want the flet build apk to succeed with the inclusion of the ready library.

**Comment by usermicrodevices** [comment]
https://github.com/usermicrodevices/fletzxing Flet control for Zxing (barcode/qrcode scanner)

**Comment by ndonkoHenri** [comment, marked as answer]
pyzbar has been added for Android and iOS: https://pypi.flet.dev/pyzbar

Give it a try and let know.
